src/main/java/org/broadinstitute/hellbender/exceptions/UserException.java

```diff
@@ -93,6 +94,26 @@ public CouldNotReadInputFile(String message) {
         }
     }
 
+    public static class OutOfDateIndex extends UserException {
+        private static final long serialVersionUID = 0L;
+
+        public OutOfDateIndex(String fileName, String indexCreationClassName) {
+            super( "Index file " + fileName + " is out of date (index older than input file). Use "
+                    + indexCreationClassName + " to make a new index."
+            );
+        }
+    }
+
+    public static class OutdatedIndexVersion extends UserException {
```

Comment:
Not sure this case deserves a separate exception -- seems like it could be subsumed under `UserException.OutOfDateIndex`, since an index with an outdated version is also "out of date", strictly speaking.

Reply:
Fixed!